joint_trajectory_controller/include/joint_trajectory_controller/trajectory.hpp

```diff
@@ -124,14 +131,28 @@ class Trajectory
   JOINT_TRAJECTORY_CONTROLLER_PUBLIC
   bool is_sampled_already() const { return sampled_already_; }
 
+  void reset_ruckig_smoothing()
+  {
+    have_previous_ruckig_output_ = false;
+  }
+
 private:
   std::shared_ptr<trajectory_msgs::msg::JointTrajectory> trajectory_msg_;
   rclcpp::Time trajectory_start_time_;
 
   rclcpp::Time time_before_traj_msg_;
   trajectory_msgs::msg::JointTrajectoryPoint state_before_traj_msg_;
 
-  bool sampled_already_ = false;
+  bool sampled_already_;
+
+  // For Ruckig jerk-limited smoothing
+  ruckig::InputParameter<ruckig::DynamicDOFs> ruckig_input_{0};
+  ruckig::OutputParameter<ruckig::DynamicDOFs> ruckig_output_{0};
+  // To avoid instability, Ruckig runs in a closed-loop fashion.
```

Comment:
What does this mean, “runs in the closed-loop fashion”? I thought that it is rather “open-loop” because input for the next calculation is the result of the previous one, or do we really get information from the hardware?

Is rucking stateless, or does it store it state in the library?

Reply:
Ok I see in the rest of the comment. I think we should reformulate this a bit to avoid misunderstanding.

Reply:
Added a colon to imply the user should keep reading --   :+1:

`// To avoid instability, Ruckig runs in a closed-loop fashion:`